### PR TITLE
add recipe for trie

### PR DESCRIPTION
**Pull request by spinoch**

### CODE REVIEW

**Comment by spinoch** [comment]
@conda-forge/help-python ready for review :) , along with #21729, #21730, #21731, #21732

File: recipes/trie/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/trie-{{ version }}.tar.gz
+  sha256: 0729a159ddef4bfa4c05478017b93a7a1af4e65e8a7b7628676bad13199ffe39
+
+build:
+  skip: true  # [py>=310]
```
Comment (BastianZim) [inline recipes/trie/meta.yaml:13]:
```suggestion
  noarch: python
```

File: recipes/trie/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python
```
Comment (BastianZim) [inline recipes/trie/meta.yaml:19]:
```suggestion
    - python >=3.8,<4
```

File: recipes/trie/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - pip
+    - setuptools-scm
+  run:
+    - python
```
Comment (BastianZim) [inline recipes/trie/meta.yaml:23]:
```suggestion
    - python >=3.8,<4
```

File: recipes/trie/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - eth-hash >=0.1.0,<1.0.0
+    - eth-utils >=2.0.0,<3.0.0
+    - hexbytes >=0.2.0
+    - rlp >=3,<4
+    - sortedcontainers >=2.1.0,<3
+    - typing-extensions >=4.0.0,<5  # [py<38]
```
Comment (BastianZim) [inline recipes/trie/meta.yaml:29]:
```suggestion
```
This keeps in noarch. Conda-forge only supports python>=3.8 for non noarch packages so we can just restrict it.

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just a couple of small things, then this is ready to go 🚀

**Pull request closed without merging**